lib/src/commonMain/kotlin/sovran.kotlin/Store.kt

```diff
@@ -226,21 +219,18 @@ class Store {
     // Removes any expired subscribers.
     private suspend fun clean() {
         sovranScope.launch(syncQueue) {
-            subscriptions.removeAll {
-                it.owner.get() == null
-            }
+            subscriptions.clear()
         }.join()
     }
 
     data class Container(var state: State)
 
     internal class Subscription<StateT : State>(
-            obj: Subscriber, val handler: Handler<StateT>,
-            val key: KClass<StateT>,
-            val queue: CoroutineDispatcher
+        val obj: Subscriber, val handler: Handler<StateT>,
+        val key: KClass<StateT>,
+        val queue: CoroutineDispatcher
     ) {
         val subscriptionID: SubscriptionID = createNextSubscriptionID()
-        val owner: WeakReference<Any> = WeakReference(obj)
```

Comment:
we need weak reference here, so sovran is not holding an object in memory forever, thus minimizing the chance of memory leak

Reply:
this is tricky, we need a solution for multiplatform and not just for java.